packages/grid/test/lit-renderer-directives.test.js

```diff
@@ -0,0 +1,281 @@
+    describe('basic', () => {
+      let cell;
+
+      beforeEach(async () => {
+        grid = await renderGrid(container, { items: ['Item'], rowDetails: 'Row Details' });
+        grid.openItemDetails(grid.items[0]);
+        cell = getRowDetailsCell(grid.$.items, { row: 0 });
+      });
+
+      it('should set `rowDetailsRenderer` property when the directive is attached', () => {
+        expect(grid.rowDetailsRenderer).to.exist;
+      });
+
+      it('should unset `rowDetailsRenderer` property when the directive is detached', async () => {
+        await renderGrid(container, { items: ['Item'] });
+        expect(grid.rowDetailsRenderer).not.to.exist;
+      });
+
+      it('should render the row details cell content with the renderer', () => {
+        expect(cell._content.textContent).to.equal('Row Details');
+      });
+
+      it('should re-render the row details cell content when a renderer dependency changes', async () => {
+        await renderGrid(container, { items: ['Item'], rowDetails: 'New Row Details' });
+        expect(cell._content.textContent).to.equal('New Row Details');
+      });
+    });
+
+    describe('arguments', () => {
+      let rendererSpy;
+
+      beforeEach(async () => {
+        rendererSpy = sinon.spy();
+        render(
+          html`<vaadin-grid .items="${['Item']}" ${gridRowDetailsRenderer(rendererSpy)}>
+            <vaadin-grid-column></vaadin-grid-column>
+          </vaadin-grid>`,
+          container,
+        );
+        grid = container.querySelector('vaadin-grid');
+        grid.openItemDetails(grid.items[0]);
+      });
+
+      it('should pass the item to the renderer', () => {
+        expect(rendererSpy.firstCall.args[0]).to.equal('Item');
+      });
+
+      it('should pass the model to the renderer', () => {
+        const cell = getRowDetailsCell(grid.$.items, { row: 0 });
+        const model = grid.__getRowModel(cell.parentElement);
+        expect(rendererSpy.firstCall.args[1]).to.deep.equal(model);
+      });
+
+      it('should pass the grid instance to the renderer', () => {
+        expect(rendererSpy.firstCall.args[2]).to.equal(grid);
+      });
+    });
+  });
+
+  describe('columnBodyRenderer', () => {
+    describe('basic', () => {
+      let cell;
+
+      beforeEach(async () => {
+        grid = await renderGrid(container, { items: ['Item'] });
+        cell = getCell(grid.$.items, { row: 0, col: 0 });
+        column = grid.querySelector('vaadin-grid-column');
+      });
+
+      it('should set `renderer` property when the directive is attached', () => {
+        expect(column.renderer).to.exist;
+      });
+
+      it('should unset `renderer` property when the directive is detached', async () => {
+        await renderGrid(container, {});
+        expect(column.renderer).not.to.exist;
+      });
+
+      it('should render the body cells content with the renderer', () => {
+        expect(cell._content.textContent).to.equal('Item');
```

Comment:
nit: could also use `getCellContent(cell).textContent` instead. Not a blocker.

Reply:
Done.